### PR TITLE
use ES modules

### PR DESCRIPTION
**Pull request by davidchambers**
sanctuary-js/sanctuary#726

### CODE REVIEW

File: .config [hunk 1]
```diff
@@ -2,4 +2,3 @@ repo-owner = sanctuary-js
 repo-name = sanctuary-show
 contributing-file = .github/CONTRIBUTING.md
 heading-level = 3
-module-type = commonjs
```
Comment (davidchambers) [inline .config:5]: `module-type` defaults to `esm` as of sanctuary-js/sanctuary-scripts#56.

File: index.js [hunk 2]
```diff
@@ -30,171 +30,152 @@
 //. };
 //. ```
 
-(f => {
+export {show as default};
```
Comment (davidchambers) [inline index.js:33]: Aldwin, are you happy to use a default export in this case or would you prefer `export {show};`?

File: package.json [hunk 3]
```diff
@@ -7,6 +7,11 @@
     "type": "git",
     "url": "git://github.com/sanctuary-js/sanctuary-show.git"
   },
+  "type": "module",
+  "exports": {
+    ".": "./index.js",
+    "./package.json": "./package.json"
+  },
```
Comment (davidchambers) [inline package.json:14]: <https://nodejs.org/api/packages.html#package-entry-points>

I want to continue to support importing from `sanctuary-show/package.json` as I consider this part of the public API.

File: test/index.js [hunk 4]
```diff
@@ -1,12 +1,7 @@
-'use strict';
+import assert from 'node:assert';
 
-const assert = require ('assert');
+import show from 'sanctuary-show';
```
Comment (davidchambers) [inline test/index.js:3]: <https://nodejs.org/api/packages.html#self-referencing-a-package-using-its-name>

File: index.js (hunk 2 above)
Comment (Avaq) [reply index.js:33]: I don't mind it

**Pull request merged**